lib/cli/src/commands/init.rs

```diff
@@ -0,0 +1,472 @@
+use cargo_metadata::{CargoOpt, MetadataCommand};
+use clap::Parser;
+use std::collections::HashMap;
+use std::path::Path;
+use std::path::PathBuf;
+
+/// CLI args for the `wasmer init` command
+#[derive(Debug, Parser)]
+pub struct Init {
+    /// Initialize wapm.toml for a library package
+    #[clap(long, name = "lib")]
+    pub lib: bool,
+    /// Initialize wapm.toml for a binary package
+    #[clap(long, name = "bin")]
+    pub bin: bool,
+    /// Initialize an empty wapm.toml
+    #[clap(long, name = "empty")]
+    pub empty: bool,
+    /// Force overwriting the wapm.toml, even if it already exists
+    #[clap(long, name = "overwrite")]
+    pub overwrite: bool,
+    /// Don't display debug output
+    #[clap(long, name = "quiet")]
+    pub quiet: bool,
+    /// Ignore the existence of cargo wapm / cargo wasmer
+    #[clap(long, name = "no-cargo-wapm")]
+    pub no_cargo_wapm: bool,
+    /// Namespace to init with, default = current logged in user or _
+    #[clap(long, name = "namespace")]
+    pub namespace: Option<String>,
+    /// Version of the initialized package
+    #[clap(long, name = "version")]
+    pub version: Option<semver::Version>,
+    /// If the `manifest-path` is a Cargo.toml, use that file to initialize the wapm.toml
+    #[clap(long, name = "manifest-path")]
+    pub manifest_path: Option<PathBuf>,
+    /// Add default dependencies for common packages (currently supported: `python`, `js`)
+    #[clap(long, name = "template")]
+    pub template: Option<String>,
```

Comment:
Can we use [the `#[derive(clap::ValueEnum)]` attribute] here? It'll let us specify the template using an enum (less string comparisons = better).

`clap` will will automatically set `possible_values = ["js", "python"]` so people can see which values are available, and it'll error out if they enter the wrong thing (possibly with a suggestion if they do a typo).

Reply:
ok done